packages/block-editor/src/components/block-heading-level-dropdown/index.js

```diff
@@ -32,14 +33,18 @@ const POPOVER_PROPS = {
  * Dropdown for selecting a heading level (1 through 6) or paragraph (0).
  *
  * @param {WPHeadingLevelDropdownProps} props Component props.
- *
- * @return {ComponentType} The toolbar.
  */
 export default function HeadingLevelDropdown( {
 	options = HEADING_LEVELS,
 	value,
 	onChange,
 } ) {
+	const blockEditingMode = useBlockEditingMode();
```

Comment:
Alternatively we move the this to context and `useBlockEditingMode` just grabs the context instead of subscribing to the store.

Reply:
This might be a more future-proof solution. Do you see any downsides to it?

Reply:
Might be good to get @noisysocks' opinion.

I think there are quite a few places that use `useBlockEditingMode` in this way so if there's a way to reduce the performance impact in a more widespread way, it might indeed be a better option.

Reply:
Ok, I created #57950. That's a better solution.